hoodie-common/pom.xml

```diff
@@ -110,10 +110,6 @@
       <artifactId>hadoop-common</artifactId>
       <classifier>tests</classifier>
     </dependency>
-    <dependency>
```

Comment:
Removing this dependency can sometimes still result in NoSuchMethod or other runtime errors elsewhere because of different versions of commons-lang3 getting pulled due to multiple "transitive" dependencies.

To avoid potential landmines, Can you check using mvn dependency:tree with this code-change to see if different versions of commons-lang3 is appearing in multiple different paths. You would need to use both mvn dependency:tree and parent dependency's pom files to figure out.

Reply:
I think lang3 dependency in hoodie-cli is coming from hoodie-utilities which will be shaded (with Vinoth's deltastreamer changes). Rest of the dependencies are only through spark-core which is provided as you mentioned. So, we should be good here.

Reply:
Yup